### PR TITLE
Use Mantine for DataSelector table picker

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Follow up after #40101.

We're getting rid of styled components wherever possible and replacing them with Mantine UI elements.
Everything should be the same after this refactor, except for the `NoTablesFound` that previously had a padding of 64px, but the largest spacing we currently have in Mantine theme is 32px, which corresponds to `xl` size.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx [hunk 1]
```diff
@@ -143,21 +133,28 @@ const DataSelectorTablePicker = ({
...
-        </Container>
+        </Box>
       </DelayGroup>
     );
   } else {
     return (
       <Section>
         <DataSelectorSectionHeader header={header} />
-        <NoTablesFound>{t`No tables found in this database.`}</NoTablesFound>
+        <Box p="xl" ta="center">{t`No tables found in this database.`}</Box>
```
Comment (nemanjaglumac) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx:143]: Before (64px)
image

After (32px)
image

If @metabase/design feels strongly about using the larger padding here, I can add it as the `style` prop.

Comment (ranquild) [reply frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx:143]: @nemanjaglumac you can always set `p="4rem"`

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx [hunk 2]
```diff
@@ -143,21 +133,28 @@ const DataSelectorTablePicker = ({
...
   }
 };
 
 const LinkToDocsOnReferencingSavedQuestionsInQueries = () => (
-  <LinkToDocsContainer>
+  <Box
+    p="md"
+    ta="center"
+    style={{
+      backgroundColor: color("bg-light"),
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx:154]: `bg` prop`, e.g. `bg="bg-light"`

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx [hunk 3]
```diff
@@ -120,7 +110,7 @@ const DataSelectorTablePicker = ({
 
     return (
       <DelayGroup>
-        <Container>
+        <Box component="span" w="300px" style={{ overflowY: "auto" }}>
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx:113]: rem

**Review by ranquild — APPROVED** [review]

**Pull request merged**